File: src/main/java/so/aporia/utils/events/impl/MouseScrollEvent.java
package so.aporia.utils.events.impl;

/** Fired when the mouse wheel is scrolled. */
public final class MouseScrollEvent {

    private final double x, y;
    private final double deltaX, deltaY;
    private boolean cancelled;

    public MouseScrollEvent(double x, double y, double deltaX, double deltaY) {
        this.x = x; this.y = y; this.deltaX = deltaX; this.deltaY = deltaY;
    }

    public double x()      { return x; }
    public double y()      { return y; }
    public double deltaX() { return deltaX; }
    /** Positive = scroll up, negative = scroll down. */
    public double deltaY() { return deltaY; }

    public void cancel()         { cancelled = true; }
    public boolean isCancelled() { return cancelled; }
}

File: src/so/aporia/utils/events/impl/MouseScrollEvent.kt
package so.aporia.utils.events.impl
import com.chaos.annotation.ChaosNative
@ChaosNative
class MouseScrollEvent(val x: Double, val y: Double, val deltaX: Double, val deltaY: Double) {

    var cancelled = false
        private set

    fun cancel() { cancelled = true }

    fun x(): Double = x
    fun y(): Double = y
    fun deltaX(): Double = deltaX
    fun deltaY(): Double = deltaY
}